sentry-rails/lib/sentry/rails/action_cable.rb

```diff
@@ -0,0 +1,72 @@
+module Sentry
+  module Rails
+    module ActionCable
+      class ErrorHandler
+        ACTION_CABLE_NAME = 'ActionCable'
+
+        def self.capture(env, transaction_name:, extra_context: nil, &block)
+          Sentry.with_scope do |scope|
+            scope.set_rack_env(env)
+            scope.set_extras(action_cable: extra_context) if extra_context
+            scope.set_transaction_name("#{ACTION_CABLE_NAME}/#{transaction_name}")
+
+            begin
+              block.call
+            rescue Exception => e # rubocop:disable Lint/RescueException
+              Sentry.capture_exception(e)
+
+              raise
+            end
+          end
+        end
+      end
+
+      module Connection
+        private
+
+        def handle_open
+          ErrorHandler.capture(env, transaction_name: "#{self.class.name}#connect") do
+            super
+          end
+        end
+
+        def handle_close
+          ErrorHandler.capture(env, transaction_name: "#{self.class.name}#disconnect") do
+            super
+          end
+        end
+      end
+
+      module Channel
+        module Subscriptions
+          def self.included(base)
+            base.class_eval do
+              set_callback :subscribe, :around, ->(_, block) { sentry_capture(:subscribed, &block) }, prepend: true
```

Comment:
why don't we use `rescue_from` but around callbacks?

Reply:
I'm definitely open to another approach here, @st0012! I went down the `around` route to allow channels/connections to set any additional context and/or capture their own exceptions, similar to the [`Sentry::Rack::CaptureExceptions`](https://github.com/getsentry/sentry-ruby/blob/7630b72817eff892eab73b5a1704abd5d2ec0c6c/sentry-ruby/lib/sentry/rack/capture_exceptions.rb) middleware.